x-pack/plugins/apm/public/components/app/service_inventory/index.tsx

```diff
@@ -122,7 +149,12 @@ export function ServiceInventory() {
     !userHasDismissedCallout &&
     shouldDisplayMlCallout(anomalyDetectionSetupState);
 
-  const isLoading = mainStatisticsStatus === FETCH_STATUS.LOADING;
+  const isLoading =
+    sortedAndFilteredServicesStatus === FETCH_STATUS.LOADING ||
+    (sortedAndFilteredServicesStatus === FETCH_STATUS.SUCCESS &&
+      sortedAndFilteredServicesData?.services.length === 0 &&
+      mainStatisticsStatus === FETCH_STATUS.LOADING);
```

Comment:
A little complex for a single loading state 😅

Reply:
I can break it up in variables, I don't think the logic itself would get easier, or do you have any suggestions?

Reply:
I don't have specific suggestions - just wondering if this is a symptom of having too complex requirements for loading states. Or if this is something that can be handled more elegantly in useFetcher.